src/main/java/fi/hsl/transitdata/tripupdate/processing/BaseProcessor.java

```diff
@@ -34,21 +35,15 @@ public BaseProcessor(EventType eventType, TripUpdateProcessor tripProcessor) {
     protected abstract PubtransData parseSharedData(Message msg) throws InvalidProtocolBufferException;
 
     @Override
-    public GtfsRealtime.TripUpdate processMessage(Message msg) {
-
-        GtfsRealtime.TripUpdate tripUpdate = null;
-
+    public Optional<GtfsRealtime.TripUpdate> processMessage(Message msg) {
```

Comment:
Do we generally use `Optional.empty()`to indicate that something went wrong (instead of throwing exceptions)?

Reply:
I've used Exceptions on fatal errors, Optionals on something that just needs to be mitigated somehow.